Yank/experiment.py

```diff
@@ -1317,129 +1615,174 @@ def _validate_systems(self, systems_description):
+                if len(error_collection) > 0:
+                    error(field, base_error + ''.join(error_collection))
+            return _region_clash_validator
+
+        def generate_is_pipeline_solvent_with_receptor_validator(system_descr, cross_id):
+
+            def _is_pipeline_solvent_with_receptor(field, solvent_id, error):
+                if cross_id in system_descr:
+                    solvent = self._db.solvents.get(solvent_id, {})
+                    if (solvent.get('nonbonded_method') != openmm.app.NoCutoff and
+                        'clearance' not in solvent):
+                        error(field, 'Explicit solvent {} does not specify clearance.'.format(solvent_id))
+            return _is_pipeline_solvent_with_receptor
+
+        # Define systems Schema
+        systems_schema_yaml = """
+        # DSL Schema
+        ligand_dsl:
+            required: no
+            type: string
+            dependencies: [phase1_path, phase2_path]
+        solvent_dsl:
+            required: no
+            type: string
+            dependencies: [phase1_path, phase2_path]
+
+        # Phase paths
+        phase1_path:
+            required: no
+            type: list
+            schema:
+                type: string
+                validator: file_exists
+            dependencies: phase2_path
+            coerce: sort_alphabetically_by_extension
+        phase2_path:
+            required: no
+            type: list
+            schema:
+                type: string
+                validator: file_exists
+            validator: is_system_files_matching_phase1
+            coerce: sort_alphabetically_by_extension
+        solvent:
+            required: no
+            type: string
+            excludes: [solvent1, solvent2]
+            allowed: SOLVENT_IDS_POPULATED_AT_RUNTIME
+            validator: PIPELINE_SOLVENT_DETERMINED_AT_RUNTIME_WITH_RECEPTOR
+            oneof:
+                - dependencies: [phase1_path, phase2_path]
+                - dependencies: [receptor, ligand]
+        solvent1:
+            required: no
+            type: string
+            excludes: solvent
+            allowed: SOLVENT_IDS_POPULATED_AT_RUNTIME
+            validator: PIPELINE_SOLVENT_DETERMINED_AT_RUNTIME_WITH_SOLUTE
+            oneof:
+                - dependencies: [phase1_path, phase2_path, solvent2]
+                - dependencies: [solute, solvent2]
+        solvent2:
+            required: no
+            type: string
+            excludes: solvent
+            allowed: SOLVENT_IDS_POPULATED_AT_RUNTIME
+            validator: PIPELINE_SOLVENT_DETERMINED_AT_RUNTIME_WITH_SOLUTE
+            oneof:
+                - dependencies: [phase1_path, phase2_path, solvent2]
+                - dependencies: [solute, solvent2]
+        gromacs_include_dir:
+            required: no
+            type: string
+            dependencies: [phase1_path, phase2_path]
+            validator: directory_exists
```

Comment:
We did lose that ability with the current setup, but I figured it would not be a large problem since the only real effect that would have is the directory goes unused for Amber and OpenMM inputs. Given that, I figured there was no harm in not checking. If you think we should, I can try to rework it.

Reply:
Works for me!